cmd/timestamp-cli/app/verify.go

```diff
@@ -130,22 +131,17 @@ func verifyArtifactWithTSR(ts *timestamp.Timestamp) error {
 		return err
 	}
 
-	h := ts.HashAlgorithm.New()
-	b := make([]byte, h.Size())
-
-	r := bytes.NewReader(artifactBytes)
-	n, err := r.Read(b)
-	if err == io.EOF {
-		return err
-	}
+	return verifyHashedMessages(ts.HashAlgorithm.New(), ts.HashedMessage, bytes.NewReader(artifactBytes))
```

Comment:
if you change line 128 to just `os.Open` then you can pass the `*os.File` as it implements io.Reader